frontend/src/app/lib/router.js

```diff
@@ -8,24 +8,27 @@ import ExperimentsListPage from '../containers/ExperimentsListPage';
 import ExperimentPage from '../containers/ExperimentPage';
 import RetirePage from '../containers/RetirePage';
 
+import App from '../components/App';
 import LegacyPage from '../components/LegacyPage';
 import NotFoundPage from '../components/NotFoundPage';
 import SharePage from '../components/SharePage';
 import ErrorPage from '../components/ErrorPage';
 import OnboardingPage from '../components/OnboardingPage';
 
 const AppRouter = ({ history }) => (
-  <Router history={history}>
-    <Route path="/" component={LandingPage} />
-    <Route path="/experiments/" component={ExperimentsListPage} />
-    <Route path="/experiments/:slug" component={ExperimentPage} />
-    <Route path="/legacy" component={LegacyPage} />
-    <Route path="/404" component={NotFoundPage} />
-    <Route path="/share" component={SharePage} />
-    <Route path="/error" component={ErrorPage} />
-    <Route path="/onboarding" component={OnboardingPage} />
-    <Route path="/retire" component={RetirePage} />
-    <Route path="*" component={NotFoundPage} />
+  <Router history={history} component={App}>
+    <Route path="/" component={App}>
```

Comment:
This is how you might use `<App>`; When matching, its `children` prop is set to any nested components that also match. This works infinitely deep, and allows for some really cool DRY patterns.